state/protocol/protocol_state/epochs/fallback_statemachine_test.go

```diff
@@ -358,28 +429,26 @@ func (s *EpochFallbackStateMachineSuite) TestEpochFallbackStateMachineInjectsMul
 			stateMachine, err := NewFallbackStateMachine(s.params, candidateView, parentProtocolState.Copy())
 			require.NoError(s.T(), err)
 
-			var (
-				nextEpochSetup  *flow.EpochSetup
-				nextEpochCommit *flow.EpochCommit
-			)
+			previousEpochSetup, previousEpochCommit := parentProtocolState.PreviousEpochSetup, parentProtocolState.PreviousEpochCommit
+			currentEpochSetup, currentEpochCommit := parentProtocolState.CurrentEpochSetup, parentProtocolState.CurrentEpochCommit
+			nextEpochSetup, nextEpochCommit := parentProtocolState.NextEpochSetup, parentProtocolState.NextEpochCommit
 			if candidateView > parentProtocolState.CurrentEpochFinalView() {
 				require.NoError(s.T(), stateMachine.TransitionToNextEpoch())
+
+				// after we have transitioned to the next epoch, we need to update the current epoch
+				// for the next iteration we can use parent protocol state again
+				previousEpochSetup, previousEpochCommit = currentEpochSetup, currentEpochCommit
+				currentEpochSetup, currentEpochCommit = parentProtocolState.NextEpochSetup, parentProtocolState.NextEpochCommit
+				nextEpochSetup, nextEpochCommit = nil, nil
 			}
 
 			updatedState, _, _ := stateMachine.Build()
 
-			if candidateView < currentEpochLastView {
-				// as the next epoch has been committed, we need to respect that in current epoch
-				// when we transition to the next epoch we will pass nil instead.
-				nextEpochSetup = parentProtocolState.NextEpochSetup
-				nextEpochCommit = parentProtocolState.NextEpochCommit
-			}
-
 			parentProtocolState, err = flow.NewRichProtocolStateEntry(updatedState,
-				parentProtocolState.PreviousEpochSetup,
-				parentProtocolState.PreviousEpochCommit,
-				parentProtocolState.CurrentEpochSetup,
-				parentProtocolState.CurrentEpochCommit,
+				previousEpochSetup,
+				previousEpochCommit,
+				currentEpochSetup,
+				currentEpochCommit,
 				nextEpochSetup,
 				nextEpochCommit)
 
```

Comment:
the lines below checks that the constructor above did not error. I think these belong very closely together 
```suggestion
```